src/helpers/gateway/mod.rs

```diff
@@ -116,6 +138,85 @@ impl<T: Transport> Gateway<T> {
                 .get_or_create(channel_id, || self.transport.receive(channel_id)),
         )
     }
+
+    fn create_idle_tracker(senders: Arc<GatewaySenders>, receivers: Arc<GatewayReceivers<T>>) -> tokio::task::JoinHandle<()> {
+        tokio::task::spawn(async move {
+            // Perform some periodic work in the background
+            loop {
+                let _ = tokio::time::sleep(Duration::from_secs(5)).await;
+                if senders.check_idle_and_reset() && receivers.check_idle_and_reset() {
+                    let sender_missing_records =senders.get_all_missing_records();
+                    let sender_message : HashMap<&ChannelId, String> = sender_missing_records.iter().map(|entry|{
+                        let (SenderStatus {next, current_written , buf_size, message_size},missing_records) = entry.1;
```

Comment:
I don't see where you use sender status struct fields, is it intentional or I am missing something?

Reply:
They are used in the next few lines.  E.g. ` let chunk_head = next - current_written / message_size;` uses `next`, etc